Source/WebCore/platform/graphics/displaylists/DisplayListRecorder.cpp

```diff
@@ -333,197 +314,94 @@ void Recorder::drawPattern(ImageBuffer& imageBuffer, const FloatRect& destRect,
-    recordDrawRect(rect, borderThickness);
-}
-
-void Recorder::drawLine(const FloatPoint& point1, const FloatPoint& point2)
-{
-    appendStateChangeItemIfNecessary();
-    recordDrawLine(point1, point2);
-}
-
-void Recorder::drawLinesForText(const FloatPoint& point, float thickness, const DashArray& widths, bool printing, bool doubleLines, StrokeStyle style)
-{
-    appendStateChangeItemIfNecessary();
-    recordDrawLinesForText(FloatPoint(), toFloatSize(point), thickness, widths, printing, doubleLines, style);
-}
-
-void Recorder::drawDotsForDocumentMarker(const FloatRect& rect, DocumentMarkerLineStyle style)
-{
-    appendStateChangeItemIfNecessary();
-    recordDrawDotsForDocumentMarker(rect, style);
-}
-
-void Recorder::drawEllipse(const FloatRect& rect)
-{
-    appendStateChangeItemIfNecessary();
-    recordDrawEllipse(rect);
-}
-
-void Recorder::drawPath(const Path& path)
-{
-    appendStateChangeItemIfNecessary();
-    recordDrawPath(path);
-}
-
-void Recorder::drawFocusRing(const Path& path, float outlineWidth, const Color& color)
-{
-    appendStateChangeItemIfNecessary();
-    recordDrawFocusRingPath(path, outlineWidth, color);
-}
-
-void Recorder::drawFocusRing(const Vector<FloatRect>& rects, float outlineOffset, float outlineWidth, const Color& color)
-{
-    appendStateChangeItemIfNecessary();
-    recordDrawFocusRingRects(rects, outlineOffset, outlineWidth, color);
-}
-
-void Recorder::fillRect(const FloatRect& rect, RequiresClipToRect requiresClipToRect)
-{
-    appendStateChangeItemIfNecessary();
-    recordFillRect(rect, requiresClipToRect);
-}
-
-void Recorder::fillRect(const FloatRect& rect, Gradient& gradient, const AffineTransform& gradientSpaceTransform, RequiresClipToRect requiresClipToRect)
-{
-    appendStateChangeItemIfNecessary();
-    recordFillRectWithGradientAndSpaceTransform(rect, gradient, gradientSpaceTransform, requiresClipToRect);
-}
-
-void Recorder::fillRect(const FloatRect& rect, const Color& color)
-{
-    appendStateChangeItemIfNecessary();
-    recordFillRectWithColor(rect, color);
-}
-
-void Recorder::fillRect(const FloatRect& rect, Gradient& gradient)
-{
-    appendStateChangeItemIfNecessary();
-    recordFillRectWithGradient(rect, gradient);
-}
-
-void Recorder::fillRect(const FloatRect& rect, const Color& color, CompositeOperator op, BlendMode blendMode)
-{
-    appendStateChangeItemIfNecessary();
-    recordFillCompositedRect(rect, color, op, blendMode);
-}
-
-void Recorder::fillRoundedRect(const FloatRoundedRect& rect, const Color& color, BlendMode blendMode)
-{
-    appendStateChangeItemIfNecessary();
-    recordFillRoundedRect(rect, color, blendMode);
-}
-
-void Recorder::fillRectWithRoundedHole(const FloatRect& rect, const FloatRoundedRect& roundedHoleRect, const Color& color)
-{
-    appendStateChangeItemIfNecessary();
-    recordFillRectWithRoundedHole(rect, roundedHoleRect, color);
-}
-
 void Recorder::fillPath(const Path& path)
 {
     appendStateChangeItemIfNecessary();
```

Comment:
Maybe this function can have only one switch statement to direct the call to GraphicsContext calls:

```
    if (auto segment = path.singleSegment()) {
#if ENABLE(INLINE_PATH_DATA)
        if (auto line = path.singleDataLine())
            fillPathLine(*line);
        else if (auto arc = path.singleArc())
            fiillPathArc(*arc);
        else if (auto closedArc = path.singleClosedArc())
            fillPathClosedArc(*closedArc);
        else if (auto curve = path.singleQuadCurve())
            fillPathQuadCurve(*curve);
        else if (auto curve = path.singleBezierCurve())
            fillPathBezierCurve(*curve);
        else
#endif
            fillPathSegment(*segment);
        return;
    }

    fillFullPath(path);
```

But I guess you plan to fix this in part 2.

Reply:
Yes. Eventually all these cases should be just removed. They're just doing the same thing, spending the lines and bringing complexity.